src/main/java/macrobase/analysis/summary/BatchSummarizer.java

```diff
@@ -0,0 +1,58 @@
+package macrobase.analysis.summary;
+
+import com.google.common.base.Stopwatch;
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.analysis.summary.itemset.FPGrowthEmerging;
+import macrobase.analysis.summary.itemset.result.ItemsetResult;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+import macrobase.util.IterUtils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class BatchSummarizer extends Summarizer {
+    private FPGrowthEmerging fpg;
+    protected final Double minSupport;
+    protected final Double minOIRatio;
+
+    public BatchSummarizer(MacroBaseConf conf, Iterator<OutlierClassificationResult> results) {
```

Comment:
Let's rename `results` to `input` here.
The reason I'm being pedantic about this is that we have an `input` in the `next()` function and it's not immediately obvious where that comes from.